translator/translator/translator.py

```diff
@@ -0,0 +1,216 @@
+"""Welcome to Reflex! This file outlines the steps to create a basic app."""
+
+import reflex as rx
+
+from datetime import datetime
+from googletrans import Translator
+from googletrans.constants import LANGUAGES
+
+
+LANGUAGE_CODE_TO_NAME = {k: v.title() for k, v in LANGUAGES.items()}
+LANGUAGE_NAMES = list(LANGUAGE_CODE_TO_NAME.values())
+
+trans = Translator()
+
+
+class Message(rx.Base):
+    original_text: str
+    text: str
+    created_at: str
+    to_lang: str
+
+
+class State(rx.State):
+    """The app state."""
+
+    text: str
+    messages: list[Message] = []
+    lang: str = LANGUAGE_CODE_TO_NAME["zh-cn"]
+
+    @rx.var
+    def input_missing(self) -> bool:
+        return not self.text.strip()
+
+    @rx.var
+    def output(self) -> str:
+        if self.input_missing:
+            return "Translations will appear here."
+        # The destination language `dest` is case insensitive, `translate` API call accepts both language names and codes.
+        translated = trans.translate(self.text, dest=self.lang)
+        return translated.text
+
+    def post(self, form_data: dict[str, str]):
+        if not (text := form_data.get("text")):
+            return
+        # Insert the message at the top of the list.
+        self.messages.insert(
+            0,
+            Message(
+                original_text=text,
+                text=self.output,
+                created_at=datetime.now().strftime("%B %d, %Y %I:%M %p"),
+                to_lang=self.lang,
+            ),
+        )
+        # Manually reset the form input, since input in this app is fully controlled, it does not reset on submit.
+        self.text = ""
+
+
+# Define views.
+
+
+def header() -> rx.Component:
+    """The header and the description."""
+    return rx.box(
+        rx.heading("Translator 🗺", size="8"),
+        rx.text(
+            "Translate things and post them as messages!",
+            color_scheme="gray",
+        ),
+        margin_bottom="1rem",
+    )
+
+
+def down_arrow() -> rx.Component:
+    return rx.vstack(
+        rx.icon(
+            tag="arrow_down",
+            color="gray",
+        ),
+    )
+
+
+def text_box(text) -> rx.Component:
+    return rx.text(
+        text,
+        background_color="white",
+        padding="1rem",
+        border_radius="8px",
+    )
+
+
+def past_translation(message: Message) -> rx.Component:
+    """A layout that contains a past translation."""
+    return rx.box(
+        rx.flex(
+            text_box(message.original_text),
+            down_arrow(),
+            text_box(message.text),
+            rx.hstack(
+                rx.text(message.to_lang),
+                rx.text("·"),
+                rx.text(message.created_at),
+                spacing="2",
+                font_size="0.8rem",
+                color="var(--gray-11)",
```

Comment:
we should use `rx.color("gray", 11)` here instead

same with other instances of explicit `var(...)` usage